src/include/btmem.h

```diff
@@ -897,37 +905,44 @@ struct __wt_ref {
 	WT_PAGE_DELETED	  *page_del;	/* Deleted page information */
 	WT_PAGE_LOOKASIDE *page_las;	/* Lookaside information */
 
-/* A macro wrapper allowing us to remember the callers code location */
-#define	WT_REF_CAS_STATE(session, ref, old_state, new_state)          \
-	__wt_ref_cas_state_int((session), (ref), (old_state), (new_state),\
-	__FILE__, __LINE__)
+	/*
+	 * In DIAGNOSTIC mode we overwrite the WT_REF on free to force failures.
+	 * Don't clear the history in that case.
+	 */
+#define	WT_REF_CLEAR_SIZE	(offsetof(WT_REF, hist))
+
 #define	WT_REF_SAVE_STATE_MAX	3
 #ifdef HAVE_DIAGNOSTIC
 	/* Capture history of ref state changes. */
 	struct __wt_ref_hist {
 		WT_SESSION_IMPL *session;
 		const char *name;
-		const char *file;
+		const char *func;
 		uint16_t line;
 		uint16_t state;
 	} hist[WT_REF_SAVE_STATE_MAX];
 	uint64_t histoff;
 #define	WT_REF_SAVE_STATE(ref, s, f, l) do {				\
 	(ref)->hist[(ref)->histoff].session = session;			\
 	(ref)->hist[(ref)->histoff].name = session->name;		\
-	(ref)->hist[(ref)->histoff].file = (f);				\
+	(ref)->hist[(ref)->histoff].func = (f);				\
 	(ref)->hist[(ref)->histoff].line = (uint16_t)(l);		\
 	(ref)->hist[(ref)->histoff].state = (uint16_t)(s);		\
 	(ref)->histoff =						\
 	    ((ref)->histoff + 1) % WT_ELEMENTS((ref)->hist);		\
 } while (0)
 #define	WT_REF_SET_STATE(ref, s) do {					\
-	WT_REF_SAVE_STATE(ref, s, __FILE__, __LINE__);			\
```

Comment:
I *think* we've been using `__FILE__` instead of `__func__ `for compatability reasons - Windows MSVC doesn't support `__func__`:
https://stackoverflow.com/questions/7008485/func-or-function-or-manual-const-char-id

Reply:
MSVC supports `__FUNCTION__`, and has since around 2003. We have:
```
/* MSVC Doesn't provide __func__, it has __FUNCTION__ */
#ifdef _MSC_VER
#define __func__        __FUNCTION__
#endif
```
in `include/msvc.h`, and we've been using it in the standard error handling function calls for awhile now.

Reply:
Ahh - I was surprised it didn't fail compilation.